pyteal/ast/abi/type.py

```diff
@@ -142,6 +142,13 @@ def _set_with_computed_type(self, value: "ComputedValue") -> Expr:
             )
         return value.store_into(self)
 
+    def __str__(self) -> str:
+        at = self.type_spec().annotation_type()
```

Comment:
this line is somewhat confusing, why not just returning `str(self.type_spec())`? Then we can guarantee that the `__str__` in type is exactly behaving the same as the one in type-spec.

Reply:
```
>>> from pyteal import *
>>> a = abi.AddressTypeSpec()
>>> str(a)
'address'
>>> a = abi.type_spec_from_annotation(abi.DynamicArray[abi.Uint32])
>>> str(a)
'uint32[]'
>>> from typing import *
>>> a = abi.type_spec_from_annotation(abi.StaticArray[abi.Uint32, Literal[10]])
>>> str(a)
'uint32[10]'
```

doubt it?

Reply:
right it spits out the abi type, which may be fine?

Reply:
yea, seems fine just tweaked it and updated tests